src/components/ProfitAndLossChart/ProfitAndLossChart.tsx

```diff
@@ -374,13 +394,14 @@ export const ProfitAndLossChart = ({
     }
   }
 
-  const CustomTooltip = ({ active, payload }: TooltipProps<number, string>) => {
-    if (active && payload && payload.length) {
-      const netProfit = payload.find(x => x.dataKey === 'netProfit')?.value ?? 0
+  const CustomTooltip = ({ active, payload: hoveredDataPoints }: TooltipProps<number, string>) => {
+    if (active && hoveredDataPoints && hoveredDataPoints.length) {
+      const dataRow = hoveredDataPoints[0]?.payload as ChartDataPoint | undefined
```

Comment:
Unfortunately we can't have Recharts enforce our types in storage, so we have to _re-enforce_ them when we read them out to render the tooltip.